internal/dao/sqlx/account_dao.go

```diff
@@ -68,6 +72,27 @@ func (di *accountDaoSqlx) GetById(ctx context.Context, id int64) (*models.Accoun
 	}
 	return result, nil
 }
+func (di *accountDaoSqlx) GetOrCreateByIdentity(ctx context.Context, orgId string, accountNumber string) (*models.Account, error) {
+	var acc, err = di.GetByOrgId(ctx, orgId)
+	var norows *dao.NoRowsError
+	if err == nil {
+		return acc, nil
+	} else if errors.As(err, &norows) {
+		ctxval.GetLogger(ctx).Info().Msgf("Account not found by org id %s", orgId)
+	} else {
+		return nil, err
+	}
+	acc, err = di.GetByAccountNumber(ctx, accountNumber)
+	if err == nil {
+		return acc, nil
+	} else if errors.As(err, &norows) {
+		ctxval.GetLogger(ctx).Info().Msgf("Account not found by account number %s", accountNumber)
```

Comment:
I suggest to pull out GetLogger into a local variable if you use it twice. But since this is an error state, probably okay I am just wondering if copy-pasting the code would not create a misleading pattern.